packages/rich-text/src/apply-format.js

```diff
@@ -92,5 +85,12 @@ export function applyFormat(
 		}
 	}
 
-	return normaliseFormats( { ...value, formats: newFormats } );
+	return normaliseFormats( {
+		...value,
+		formats: newFormats,
+		// Always revise active formats. This serves as a placeholder for new
+		// inputs with the format so new input appears with the format applied,
+		// and ensures a format of the same type uses the latest values.
+		activeFormats: [ format, ...activeFormats ],
```

Comment:
I wonder if order could matter here. Could be not just copy the formats at the selection start? Or perhaps use `getActiveFormats`?

Reply:
In my testing, it did matter: Where due to the behavior of `normaliseFormats`, the 'latest' format must occur first, since `normaliseFormats` defers to the previous entry:

https://github.com/WordPress/gutenberg/blob/4108e37d3771b5871cd720216432b5bbd5255551/packages/rich-text/src/normalise-formats.js#L27-L29

I suppose I ought to document this more clearly.

Reply:
Although looking closer at `isFormatEqual`, it's odd that it would have made any effective difference for me... considering that "latest" shouldn't matter if the attribute values (i.e. `target="_blank"` was not the same between formats anyways).